packages/daemon/src/services/index.ts

```diff
@@ -289,31 +293,80 @@ export const handleVertexAccepted = async (context: Context, _event: Event) => {
       // for the addresses present on the tx, check if there are any wallets associated
       const addressWalletMap: StringMap<Wallet> = await getAddressWalletInfo(mysql, Object.keys(addressBalanceMap));
 
-      // for each already started wallet, update databases
       const seenWallets = new Set();
-      for (const wallet of Object.values(addressWalletMap)) {
-        const walletId = wallet.walletId;
+      const addressesPerWallet = Object.entries(addressWalletMap).reduce(
+        (result: StringMap<{ addresses: string[], walletDetails: Wallet }>, [address, wallet]: [string, Wallet]) => {
+        const { walletId } = wallet;
 
-        // this map might contain duplicate wallet values, as 2 different addresses might belong to the same wallet
-        if (seenWallets.has(walletId)) continue;
         seenWallets.add(walletId);
```

Comment:
Do you need to keep this variable? You could just use the keys of `addressesPerWallet`, right?